playwright/_impl/_network.py

```diff
@@ -285,10 +285,13 @@ async def _race_with_page_close(self, future: Coroutine) -> None:
             # When page closes or crashes, we catch any potential rejects from this Route.
             # Note that page could be missing when routing popup's initial request that
             # does not have a Page initialized just yet.
+            fut = asyncio.create_task(future)
             await asyncio.wait(
-                [asyncio.create_task(future), page._closed_or_crashed_future],
+                [fut, page._closed_or_crashed_future],
                 return_when=asyncio.FIRST_COMPLETED,
             )
+            if page._closed_or_crashed_future.done():
```

Comment:
```suggestion
            if not fut.done():
```
does the same but is more readable, wdyt?

Reply:
this fut is racing with page closed event, so check the page is closed instead of if the `fut` is complete maybe more clear, I think.